wdl-format/src/config/builder.rs

```diff
@@ -30,37 +39,62 @@ pub type Result<T> = std::result::Result<T, Error>;
 pub struct Builder {
     /// The number of characters to indent.
     indent: Option<Indent>,
+    /// The maximum line length.
+    max_line_length: Option<MaxLineLength>,
 }
 
 impl Builder {
     /// Creates a new builder with default values.
-    pub fn new() -> Self {
-        Default::default()
+    pub fn new(indent: Option<Indent>, max_line_length: Option<MaxLineLength>) -> Result<Self> {
+        let indent = indent.unwrap_or_default();
+        let max_line_length = max_line_length.unwrap_or_default();
+        Ok(Self::default()
+            .indent(indent)?
+            .max_line_length(max_line_length))
     }
 
     /// Sets the indentation level.
     ///
-    /// # Notes
-    ///
     /// This silently overwrites any previously provided value for the
     /// indentation level.
-    pub fn indent(mut self, indent: Indent) -> Self {
+    pub fn indent(mut self, indent: Indent) -> Result<Self> {
+        if let Indent::Spaces(n) = indent {
+            if n > MAX_SPACE_INDENT {
+                return Err(Error::Invalid("indent"));
+            }
+        }
         self.indent = Some(indent);
+        Ok(self)
+    }
+
+    /// Sets the maximum line length.
+    ///
+    /// This silently overwrites any previously provided value for the maximum
+    /// line length.
+    pub fn max_line_length(mut self, max_line_length: MaxLineLength) -> Self {
```

Comment:
This should also check the max line length after you move the checks to the `new()` method.